From 76a0c49851b905466fa39fb62ef0119985f7cd01 Mon Sep 17 00:00:00 2001
From: Kishan Kava <kishan.kava@analog.com>
Date: Mon, 16 Feb 2026 16:30:27 +0530
Subject: [PATCH] overlay: AR0234 + Dual ADSD3500: Move i2c@3180000 node inside
 cam_i2cmux

	* Move i2c@3180000 node inside cam_i2cmux for correct I2C multiplexing.
	* Fixes issue where only 60 FPS mode worked on Arducam; other FPS modes
	  now functional.

Signed-off-by: Kishan Kava <kishan.kava@analog.com>
---
 ...7-camera-p3768-dual-adsd3500-adsd3100-arducam-ar0234.dts | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/overlay/tegra234-p3767-camera-p3768-dual-adsd3500-adsd3100-arducam-ar0234.dts b/overlay/tegra234-p3767-camera-p3768-dual-adsd3500-adsd3100-arducam-ar0234.dts
index 78dcba0..e510b3d 100644
--- a/overlay/tegra234-p3767-camera-p3768-dual-adsd3500-adsd3100-arducam-ar0234.dts
+++ b/overlay/tegra234-p3767-camera-p3768-dual-adsd3500-adsd3100-arducam-ar0234.dts
@@ -159,9 +159,6 @@
 						};
 					};
 				};
-				i2c@3180000 {
-					clock-frequency = <1000000>;
-				};
 
 				cam_i2cmux {
 					status = "okay";
@@ -170,6 +167,9 @@
 					#size-cells = <0>;
 					mux-gpios = <&gpio_aon CAM_I2C_MUX GPIO_ACTIVE_HIGH>;
 					i2c-parent = <&cam_i2c>;
+					i2c@3180000 {
+						clock-frequency = <1000000>;
+					};
 					i2c@0 {
 						status = "okay";
 						reg = <0>;
-- 
2.25.1

